### PR TITLE
Exercises and Assignment for Lesson 3

### PR DESCRIPTION
**Pull request by bplanica**
Exercises and Assignment for Lesson 3 - Please let me know if anything could be done better. Thanks :)

### CODE REVIEW

File: students/bplanica/lesson02/Fizz_Buzz.py [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+while i <= 100:
+    if (i/3).is_integer() and (i/5).is_integer():  # If divisible by both 3 and 5...
+        print("FizzBuzz")
+    elif (i/3).is_integer():  # If divisible by 3...
+        print("Fizz")
+    elif (i/5).is_integer():  # If divisible by 5...
+        print("Buzz")
+    else:
+        print(i)  # ...Print counter/number
+    i += 1
```
Comment (charlesrobison) [inline students/bplanica/lesson02/Fizz_Buzz.py]: This works :)

File: students/bplanica/lesson02/Grid_Printer.py [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+    print(plus)  # print a top/bottom line
+    i = 1
+    while i <= x:  # Do until the counter = the number of rows desired
+        j = 1
+        while j <= y:
+            print(bar)  # Print as many bar lines as there are "-" in the top/bottom line
+            j += 1
+        print(plus)  # Print correcponding top/bottom line and repeat
+        i += 1
+
```
Comment (charlesrobison) [inline students/bplanica/lesson02/Grid_Printer.py]: Good!

File: students/bplanica/lesson02/Series.py [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+
+    # test if sum_series works for arbitrary initial values
+    assert sum_series(0, 3, 2) == 3
+    assert sum_series(1, 3, 2) == 2
+    assert sum_series(2, 3, 2) == 5
+    assert sum_series(3, 3, 2) == 7
+    assert sum_series(4, 3, 2) == 12
+    assert sum_series(5, 3, 2) == 19
+
+    print("tests passed!")
```
Comment (charlesrobison) [inline students/bplanica/lesson02/Series.py]: Looks good - nice assert statements!

File: students/bplanica/lesson03/list_lab.py [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+if response.upper() == "Y":
+    myList = ["Apples", "Pears", "Oranges", "Peaches"]
+    print_list(myList, "The current list items are:")
+
+    response = input("Name a fruit would you like added to the end of this list: ")
+    print()
+    myList.append(response)
+    print_list(myList, "The current list items are:")
+
+    response = int(input("Which item (number) in the list of fruits would you like to view?: "))
```
Comment (charlesrobison) [inline students/bplanica/lesson03/list_lab.py]: You can use len() here to let the user know the length of the list

File: students/bplanica/lesson03/list_lab.py [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+    myList = [response] + myList
+    print_list(myList, "The current list items are:")
+
+    response = input("Name another fruit you would like added to the beginning of the list?: ")
+    print()
+    myList.insert(0, response)
+    print_list(myList, "The current list items are:")
+
+    for i in myList:
+        if i[0].upper() == "P":
```
Comment (charlesrobison) [inline students/bplanica/lesson03/list_lab.py]: Nice use of upper()

File: students/bplanica/lesson03/list_lab.py [hunk 6]
```diff
@@ -0,0 +1,117 @@
...
+
+    for item in myList:
+        response = input("Do you like " + item + "? (Y/N)").upper()
+        while response != "Y" and response != "N":
+            response = input("Please enter Y or N: ").upper()
+        if response == "Y":
+            sub_List.append(item)
+
+    print_list(sub_List, "Here is a list of what you like!")
+    sub_List = []
```
Comment (charlesrobison) [inline students/bplanica/lesson03/list_lab.py]: Good!

File: students/bplanica/lesson03/list_lab.py [hunk 7]
```diff
@@ -0,0 +1,117 @@
...
+    response= input("Which other fruit would you like to remove?: ")
+    for item in myList:
+        if item.upper() == response.upper():
+            myList.remove(item)
+    print_list(myList, "Here is your updated list: ")
+
+
+# ----- Series 3 ----- #
+
+response = input("Run Series 3? (Y/N: ")
```
Comment (charlesrobison) [inline students/bplanica/lesson03/list_lab.py]: Looks like your missing your closing parentheses here after Y/N:

File: students/bplanica/lesson03/mailroom.py [hunk 8]
```diff
@@ -0,0 +1,144 @@
...
+# ------------------------------ #
+# Assignment 3 (Mailroom Part 1) for Python 210
+# Dev: Breeanna Planica
+# ChangeLog: (who, when, what)
+#   BPA, 8/11/2019, Created and tested script
+# ------------------------------ #
+
+# ----- DATA ----- #
+# ---------------- #
+DonorLst = [["Bill Gates", 326892.24, 326892.24],
```
Comment (charlesrobison) [inline students/bplanica/lesson03/mailroom.py]: Just like function names, these should be snake_case as well: donor_list

File: students/bplanica/lesson03/mailroom.py [hunk 9]
```diff
@@ -0,0 +1,144 @@
...
+["Mark Zuckerberg", 5465.39, 5465.39, 5465.39], 
+["Jeff Bezo", 877.33], 
+["Paul Allen", 236.14, 236.14, 236.14], 
+["M Bezo", 110463.25]] # holds original data set to start with
+
+strMenu = ("""
```
Comment (charlesrobison) [inline students/bplanica/lesson03/mailroom.py]: Same here

File: students/bplanica/lesson03/mailroom.py [hunk 10]
```diff
@@ -0,0 +1,144 @@
...
+    from operator import itemgetter
+    for row in sorted(sub_List, key=itemgetter(1), reverse=True): # sort and print in a fixed length format
+        print("{:<30}|{:>15}|{:>10}|{:>15}".format(row[0], row[1], row[2], row[3]))
+
+
+# ----- PRESENTATION ----- #
+# ------------------------ #
+
+if __name__ == '__main__':
+    main() # run main
```
Comment (charlesrobison) [inline students/bplanica/lesson03/mailroom.py]: This works well - nice job

File: students/bplanica/lesson03/slicing.py [hunk 11]
```diff
@@ -0,0 +1,54 @@
...
+assert every_other(a_string) == 'hr sasrn'
+assert every_other(a_tuple) == (1, 3, 5)
+assert every_other_middle(a_string) ==  ' sas'
+assert every_other_middle(a_tuple) == ()
+assert reversed(a_string) == 'gnirts a si ereh'
+assert reversed(a_tuple) == (5, 4, 3, 2, 1)
+assert last_first_middle(a_string) == 'stringhere is a '
+assert last_first_middle(a_tuple) == (4, 5, 1, 2, 3)
+
+print("Passed!")
```
Comment (charlesrobison) [inline students/bplanica/lesson03/slicing.py]: Looks good!

File: students/bplanica/lesson03/strformat_lab.py [hunk 12]
```diff
@@ -0,0 +1,50 @@
...
+print(f"The weight of an {myList[0]:s} is {myList[1]:.1f} and the weight of a {myList[2]:s} is {myList[3]:.1f}")
+
+myList = ["orange", 1.3, "lemon", 1.1]
+print(f"The weight of an {myList[0].upper():s} is {myList[1]*1.2:.1f} and the weight of a {myList[2].upper():s} is {myList[3]*1.2:.1f}")
+
+# ----- Task 6 ----- #
+myList = [("John", "40", "10,000"), ("Jane", "34", "575"), ("Samantha", "23", "15,575")]
+
+for tpl in myList:
+    print("{:<20}{:>3}{:>10}".format(tpl[0], tpl[1], tpl[2]))
```
Comment (charlesrobison) [inline students/bplanica/lesson03/strformat_lab.py]: Great!

File: students/bplanica/lesson03/strformat_lab.py [hunk 13]
```diff
@@ -0,0 +1,50 @@
...
+
+# ------------------------------ #
+# String Lab Assignment for Python 210
+# Dev: Breeanna Planica
+# ChangeLog: (who, when, what)
+#   BPA, 8/9/2019, Created and tested script
+# ------------------------------ #
+
+# ----- Task 1 ----- #
+myTuple = (2, 123.4567, 10000, 12345.67)
```
Comment (charlesrobison) [inline students/bplanica/lesson03/strformat_lab.py]: snake_case on all of these...

**Comment by natasha-aleksandrova** [comment]
@bplanica please address merge conflicts (sync with master)

contact is if you need help

**Comment by bplanica** [comment]
Hello! I think I addressed the conflicts, but I am not super comfortable with git yet, so please let me know if I didn't address it properly.

**Comment by natasha-aleksandrova** [comment]
@bplanica nice 👍 

no more conflicts and we can merge this now.

**Pull request merged**